src/Revisionable/Document/Repository/RevisionRepository.php

```diff
@@ -0,0 +1,182 @@
+    {
+        $documentWrapper = new MongoDocumentWrapper($document, $this->getDocumentManager());
+
+        $documentMetadata = $documentWrapper->getMetadata();
+        $documentId = (string) $documentWrapper->getIdentifier(false, true);
+        $documentClass = $documentMetadata->getName();
+
+        $qb = $this->createQueryBuilder();
+        $qb->field('revisionableId')->equals($documentId);
+        $qb->field('revisionableClass')->equals($documentClass);
+        $qb->field('version')->lte($version);
+        $qb->sort('version', 'ASC');
+
+        $revisions = $qb->getQuery()->getIterator()->toArray();
+
+        if ([] === $revisions) {
+            throw new UnexpectedValueException(sprintf('Could not find any revisions for version %d of document %s.', $version, $documentClass));
+        }
+
+        $data = [[]];
+
+        while ($revision = array_shift($revisions)) {
+            $data[] = $revision->getData();
+        }
+
+        $data = array_merge(...$data);
+
+        $this->fillDocument($document, $data);
+    }
+
+    /**
+     * Fills a document's versioned fields with the given data
+     *
+     * @param T                    $document
+     * @param array<string, mixed> $data
+     */
+    protected function fillDocument(object $document, array $data): void
+    {
+        $documentWrapper = new MongoDocumentWrapper($document, $this->getDocumentManager());
+
+        $documentMeta = $documentWrapper->getMetadata();
+
+        assert($documentMeta instanceof ClassMetadata);
+
+        $config = $this->getListener()->getConfiguration($this->getDocumentManager(), $documentMeta->getName());
+        $fields = $config['versioned'];
+
+        foreach ($data as $field => $value) {
+            if (!in_array($field, $fields, true)) {
+                continue;
+            }
+
+            $mapping = $documentMeta->getFieldMapping($field);
+
+            // Fill the embedded document
+            if ($documentWrapper->isEmbeddedAssociation($field)) {
+                if (!empty($value)) {
+                    assert(class_exists($mapping['targetDocument']));
+
+                    $embeddedMetadata = $this->getDocumentManager()->getClassMetadata($mapping['targetDocument']);
+                    $document = $embeddedMetadata->newInstance();
+                    $this->fillDocument($document, $value);
+                    $value = $document;
+                }
+            } elseif ($documentMeta->isSingleValuedAssociation($field)) {
+                assert(class_exists($mapping['targetDocument']));
+
+                $value = $value ? $this->getDocumentManager()->getReference($mapping['targetDocument'], $value) : null;
+            } else {
+                $value = $value ? $documentWrapper->convertToPHPValue($value, $documentMeta->getTypeOfField($field)) : null;
+            }
+
+            $documentWrapper->setPropertyValue($field, $value);
+            unset($fields[$field]);
+        }
+
+        /*
```

Comment:
What is the intention for this comment?

Reply:
It's been there ever since https://github.com/doctrine-extensions/DoctrineExtensions/pull/1212 was accepted in the log entry repository.  The ORM repository throws when it couldn't process all fields, no idea why the MongoDB repository has the check commented out.

Reply:
For my reference (this is a bug in loggable and this implementation)

Adding a `dump('fields list start', $fields);` after `$fields = $config['versioned']` gives:

```sh
1 "fields list start"
2 array:5 [
  0 => "subject"
  1 => "message"
  2 => "writtenAt"
  3 => "article"
  4 => "author"
]
```

And `dump('fields list end', $fields);` before the commented block:

```sh
1 "fields list end"
2 array:5 [
  0 => "subject"
  1 => "message"
  2 => "writtenAt"
  3 => "article"
  4 => "author"
]
```

`unset($fields[$field]);` can't work because the fields list isn't keyed by name.  The ORM repository uses `unset($fields[array_search($field, $fields, true)]);` so this change needs to be tested for both.

Reply:
The revisionable repository's updated with this change and the commented code in the loggable repository is removed (funny enough the loggable ORM repository also had this commented out, at this point it's just easier to remove the commented code and not try to enable it there and just ship this with it enabled).